src/components/AccountContainer.js

```diff
@@ -13,48 +13,51 @@ class AccountContainer extends Component {
 
     this.state = {
       searchTerm: '',
-      transactions: [
-        {
-          id: 1,
-          posted_at: "2017-02-28 11:00:00",
-          description: "Leather Pants, Gap co.",
-          category: "Fashion",
-          amount: -20000
-        },
-        {
-          id: 2,
-          posted_at: "2017-02-29 10:30:00",
-          description: "Paycheck from Bob's Burgers",
-          category: "Income",
-          amount: 100000
-        },
-        {
-          id: 3,
-          posted_at: "2017-05-24 10:53:00",
-          description: "'Pair Programming Illuminated' by Laurie Williams and Robert Kessler",
-          category: "Book",
-          amount: 1498
-        },
-        {
-          id: 4,
-          posted_at: "2017-05-24 08:52:00",
-          description: "Medium Iced Cold Brew, Gregory's Coffee",
-          category: "Coffee",
-          amount: 365
-        }
-      ]
+      transactions: []
     }
+    this.fetchTransactions = this.fetchTransactions.bind(this)
+    this.handleChange = this.handleChange.bind(this)
+    this.filterTransactions = this.filterTransactions.bind(this)
+    this.termCheck = this.termCheck.bind(this)
+  }
+
+  componentDidMount(){
+    this.fetchTransactions()
+  }
+
+  fetchTransactions(){
+    fetch('https://boiling-brook-94902.herokuapp.com/transactions')
+      .then( res => res.json() )
+      .then( data => this.setState({ transactions: data}) )
   }
 
   handleChange(event) {
-    // your code here
+    this.setState( {searchTerm: event.target.value})
+    // this.termCheck(this.state.searchTerm, event.target.value)
+    this.filterTransactions(this.state.searchTerm)
   }
 
-  render() {
+  filterTransactions(searchTerm){
+    let filteredTrans = this.state.transactions.filter( (trans) =>
+      trans.category.toLowerCase().includes(searchTerm.toLowerCase()) ||
+      trans.description.toLowerCase().includes(searchTerm.toLowerCase())
+      )
+    this.setState({
+      transactions: filteredTrans
```

Comment:
We discussed this in the review - the problem here is you're replacing all your transactions with the filtered list, thus losing all of your transactions. Better not to store the `filteredList` as a piece of state - just calculate it and pass it down.